File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.java
class Solution {
    public int strStr(String haystack, String needle) {
        
        int count=0,index=-1;
        
        for(int i=0;i<=haystack.length()-needle.length();i++){
            
            index=i;
            
            for(int j=0;j<needle.length();j++)
            {
                
                if(haystack.charAt(i+j)==needle.charAt(j))
                    
                    count++;
            }
            if(count==needle.length())
                return index;
            else
                count=0;       
            }
       return -1;
        }
    }

File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.kt
class Solution {
    fun strStr(haystack: String, needle: String): Int {
        
        for(i in 0 .. haystack.length-needle.length){
            var count = 0
            for(j in i until i+needle.length){
                if(haystack[j] == needle[count]) count++
                else break
            }
            if(count==needle.length) return i
        }

        return -1
    }
}